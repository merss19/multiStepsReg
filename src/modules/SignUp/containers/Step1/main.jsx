"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var react_redux_1 = require("react-redux");
var redux_form_1 = require("redux-form");
var react_router_dom_1 = require("react-router-dom");
var ducks_1 = require("./ducks");
var api_1 = require("./api");
var InputText_1 = require("components/InputText");
var RadioInput_1 = require("components/RadioInput");
var CheckboxInput_1 = require("components/CheckboxInput");
var Button_1 = require("components/Button");
var storage_1 = require("tools/storage");
var const_1 = require("../../const");
var CustomSteps_1 = require("../CustomSteps");
var CustomModal_1 = require("modules/CustomModal");
var ducks_2 = require("modules/Login/ducks");
var formValidations = require("tools/validations");
var ModalHOC_1 = require("components/HOC/ModalHOC");
var StepOneComponent = /** @class */ (function (_super) {
    __extends(StepOneComponent, _super);
    function StepOneComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    StepOneComponent.prototype.shouldComponentUpdate = function (nextProps) {
        return (nextProps.step !== this.props.step);
    };
    StepOneComponent.prototype.onSubmit = function (data) {
        var _this = this;
        this.props.userCreate(data, function (data) { return _this.resultAction(data); });
    };
    StepOneComponent.prototype.resultAction = function (data) {
        var _a = this.props, changeStep = _a.changeStep, setToken = _a.setToken;
        if (data && data.authToken) {
            setToken(data.authToken);
            storage_1.storage.save('token', data.authToken, { path: '/', maxAge: 60 * 60 * 24 * 365 * 10 });
            storage_1.storage.save('promoName', data.promoName, { path: '/', maxAge: 60 * 60 * 24 * 365 * 10 });
            changeStep(const_1.Steps.two);
        }
    };
    StepOneComponent.prototype.render = function () {
        var _this = this;
        var handleSubmit = this.props.handleSubmit;
        return (<div>
        {this.renderHeader()}
        <form className='entry__box' onSubmit={handleSubmit(function (data) { return _this.onSubmit(data); })}>
          <redux_form_1.Field name='email' placeholder='Ваш e-mail' type='text' component={InputText_1.default}/>
          <redux_form_1.Field name='password' placeholder='Ваш пароль' type='password' component={InputText_1.default}/>
          <redux_form_1.Field name='passwordAgain' placeholder='Пароль повторно' type='password' component={InputText_1.default}/>

          <div className='gender'>
            <div className='gender__group'>
              <p className='gender__title'>Ваш пол:</p>
              <redux_form_1.Field name='gender' id='male' value='male' type='radio' title='Мужчина' validate={formValidations.validateGender} component={RadioInput_1.default}/>
              <redux_form_1.Field name='gender' id='female' value='female' type='radio' title='Женщина' validate={formValidations.validateGender} component={RadioInput_1.default}/>
            </div>
            <hr className='gender__hr'/>
          </div>

          <div className='checkboxes'>
            <redux_form_1.Field name='accept' title='Принять условия ' id='accept' component={CheckboxInput_1.default}/>
          </div>

          <Button_1.Button type='submit' styleBtn={Button_1.ButtonTypes.info} wide={true}>
            Зарегистрироваться
          </Button_1.Button>


          <div className='entry__link text-center mtb20'>
            <div className='entry-nav__item'>
              <react_router_dom_1.Link to='/'>Войти</react_router_dom_1.Link>
            </div>
            <div className='entry-nav__item'>
              <react_router_dom_1.Link to='/restore'>Забыли пароль?</react_router_dom_1.Link>
            </div>
          </div>
        </form>
      </div>);
    };
    return StepOneComponent;
}(CustomSteps_1.default));
exports.StepOneComponent = StepOneComponent;
exports.StepOne = react_redux_1.connect(null, {
    toggleModal: CustomModal_1.toggleModal,
    setError: ducks_1.setError,
    userCreate: ducks_1.userCreate,
    setToken: ducks_2.setToken
})(StepOneComponent);
exports.default = ModalHOC_1.modalHOC(redux_form_1.reduxForm({
    form: 'StepOne',
    validate: formValidations.validateSignUp,
    asyncValidate: api_1.asyncValidate,
    asyncBlurFields: ['email']
})(exports.StepOne), ducks_1.moduleName, ducks_1.subModuleName);
