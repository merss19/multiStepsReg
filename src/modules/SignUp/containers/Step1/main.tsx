import * as React from 'react';
import {connect} from 'react-redux';
import {Field, FormProps, reduxForm} from 'redux-form';
import {Link} from 'react-router-dom';
import {setError, userCreate, moduleName, subModuleName} from './ducks';
import {asyncValidate} from './api';
import InputText from 'components/InputText';
import RadioInput from 'components/RadioInput';
import CheckboxInput from 'components/CheckboxInput';
import {SignUpFormData} from '../../interfaces';
import {Button, ButtonTypes} from 'components/Button';
import {storage} from 'tools/storage';
import {Steps} from '../../const';
import CustomSteps, {CustomStepsProps} from '../CustomSteps';
import {toggleModal} from "modules/CustomModal";
import {setToken} from 'modules/Login/ducks'
import * as formValidations from 'tools/validations'
import {modalHOC} from "components/HOC/ModalHOC";

interface OwnProps extends CustomStepsProps, FormProps<{}, {}, {}> {
  changeStep: (data: Steps) => void
  step: Steps
}

type ConnectedState = {
  isFetching: boolean
  isError: boolean
  errMsg: string
}
type ConnectedDispatch = {
  getPrograms(): void
  setToken(token: string): void
  toggleModal(data: boolean): void
  setError(data: boolean): any
  userCreate(data: SignUpFormData, cb: (data: any) => void): any
}

type Props = OwnProps & ConnectedDispatch;

export class StepOneComponent extends CustomSteps<Props> {

  shouldComponentUpdate(nextProps: Props) {
    return (
      nextProps.step !== this.props.step
    );
  }

  onSubmit(data: SignUpFormData) {
    this.props.userCreate(data, (data) => this.resultAction(data));
  }

  resultAction(data: any): void {
    const {changeStep, setToken} = this.props;
    if (data && data.authToken) {
      setToken(data.authToken);
      storage.save('token', data.authToken,
        {path: '/', maxAge: 60 * 60 * 24 * 365 * 10});
      storage.save('promoName', data.promoName,
        {path: '/', maxAge: 60 * 60 * 24 * 365 * 10});
      changeStep(Steps.two);
    }
  }

  render() {
    const {handleSubmit} = this.props;
    return (
      <div>
        {this.renderHeader()}
        <form className='entry__box' onSubmit={handleSubmit((data: any) => this.onSubmit(data))}>
          <Field
            name='email'
            placeholder='Ваш e-mail'
            type='text'
            component={InputText}
          />
          <Field
            name='password'
            placeholder='Ваш пароль'
            type='password'
            component={InputText}
          />
          <Field
            name='passwordAgain'
            placeholder='Пароль повторно'
            type='password'
            component={InputText}
          />

          <div className='gender'>
            <div className='gender__group'>
              <p className='gender__title'>Ваш пол:</p>
              <Field
                name='gender'
                id='male'
                value='male'
                type='radio'
                title='Мужчина'
                validate={formValidations.validateGender}
                component={RadioInput}
              />
              <Field
                name='gender'
                id='female'
                value='female'
                type='radio'
                title='Женщина'
                validate={formValidations.validateGender}
                component={RadioInput}
              />
            </div>
            <hr className='gender__hr'/>
          </div>

          <div className='checkboxes'>
            <Field
              name='accept'
              title='Принять условия '
              id='accept'
              component={CheckboxInput}/>
          </div>

          <Button
            type='submit'
            styleBtn={ButtonTypes.info}
            wide={true}
          >
            Зарегистрироваться
          </Button>


          <div className='entry__link text-center mtb20'>
            <div className='entry-nav__item'>
              <Link to='/'>Войти</Link>
            </div>
            <div className='entry-nav__item'>
              <Link to='/restore'>Забыли пароль?</Link>
            </div>
          </div>
        </form>
      </div>
    );
  }
}

export const StepOne = connect(null, {
  toggleModal,
  setError,
  userCreate,
  setToken
})(StepOneComponent);

export default modalHOC(reduxForm({
  form: 'StepOne',
  validate: formValidations.validateSignUp,
  asyncValidate,
  asyncBlurFields: ['email']
})(StepOne), moduleName, subModuleName);