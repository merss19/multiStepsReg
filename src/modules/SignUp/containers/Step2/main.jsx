"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var react_redux_1 = require("react-redux");
var CustomSteps_1 = require("../CustomSteps");
var react_router_dom_1 = require("react-router-dom");
var classNames = require("classnames");
var storage_1 = require("tools/storage");
var const_1 = require("../../const");
var Button_1 = require("components/Button");
var CustomModal_1 = require("modules/CustomModal");
var ModalHOC_1 = require("components/HOC/ModalHOC");
var ducks_1 = require("./ducks");
var selectors_1 = require("../../selectors");
var mapStateToProps = function (state) { return ({
    programs: selectors_1.selectPrograms(state),
    choosenProgram: selectors_1.selectChoosenProgram(state),
    choosenPackageType: selectors_1.selectPackageType(state),
    packages: selectors_1.selectPackages(state),
    promo: selectors_1.selectPromo(state),
    currentPackage: selectors_1.selectPackage(state),
    promoError: selectors_1.selectPromoError(state)
}); };
var StepTwoComponent = /** @class */ (function (_super) {
    __extends(StepTwoComponent, _super);
    function StepTwoComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    StepTwoComponent.prototype.componentWillMount = function () {
        var _a = this.props, programs = _a.programs, getPrograms = _a.getPrograms, packages = _a.packages, setChoosenProgram = _a.setChoosenProgram, choosenProgram = _a.choosenProgram, getPackage = _a.getPackage;
        if (!packages) {
            getPackage();
        }
        if (!programs) {
            getPrograms();
        }
    };
    StepTwoComponent.prototype.componentWillReceiveProps = function (nextProps) {
        var _a = this.props, programs = _a.programs, choosenProgram = _a.choosenProgram, setChoosenProgram = _a.setChoosenProgram;
        if (nextProps.programs) {
            if (!choosenProgram) {
                if (nextProps.programs) {
                    nextProps.programs.map(function (program) {
                        if (program.get('id') % const_1.ProgramNum.count === const_1.ProgramNum.hero) {
                            setChoosenProgram(program.get('id'));
                        }
                    });
                }
            }
        }
    };
    StepTwoComponent.prototype.shouldComponentUpdate = function (nextProps) {
        return (nextProps.step !== this.props.step
            || nextProps.packages !== this.props.packages
            || nextProps.programs !== this.props.programs
            || nextProps.choosenProgram !== this.props.choosenProgram
            || nextProps.choosenPackageType !== this.props.choosenPackageType
            || nextProps.promo !== this.props.promo
            || nextProps.currentPackage !== this.props.currentPackage
            || nextProps.promoError !== this.props.promoError);
    };
    StepTwoComponent.prototype.getPromo = function () {
        var _a = this.props, getPackage = _a.getPackage, setPromo = _a.setPromo;
        if (!this.promoText || !this.promoText.value.length) {
            return;
        }
        getPackage(this.promoText.value);
        setPromo(this.promoText.value);
    };
    StepTwoComponent.prototype.paymentCreate = function () {
        var _this = this;
        var _a = this.props, choosenProgram = _a.choosenProgram, choosenPackageType = _a.choosenPackageType, paymentCreate = _a.paymentCreate, promo = _a.promo;
        var payload = {
            authToken: storage_1.storage.load('token'),
            data: {
                program: choosenProgram,
                package: choosenPackageType,
                promoName: promo
            }
        };
        paymentCreate(payload, function () { return _this.resultAction(); });
    };
    StepTwoComponent.prototype.resultAction = function () {
        var _a = this.props, changeStep = _a.changeStep, choosenProgram = _a.choosenProgram;
        storage_1.storage.save('program', choosenProgram, { path: '/', maxAge: 60 * 60 * 24 * 365 * 10 });
        changeStep(const_1.Steps.three);
    };
    StepTwoComponent.prototype.renderPrograms = function () {
        var _a = this.props, programs = _a.programs, setChoosenProgram = _a.setChoosenProgram;
        if (!programs) {
            return <div>Нет программ</div>;
        }
        else {
            return (programs.map(function (program) { return (<li key={program.get('id')} onClick={function () { setChoosenProgram(program.get('id')); }} className={classNames('options__item', {
                'is-active': program.get('isActive'),
                'g-hero': program.get('id') % const_1.ProgramNum.count == const_1.ProgramNum.hero,
                'g-mather': program.get('id') % const_1.ProgramNum.count == const_1.ProgramNum.mother,
                'g-extreme': program.get('id') % const_1.ProgramNum.count == const_1.ProgramNum.extreme,
                'g-tomorrow': program.get('id') % const_1.ProgramNum.count == const_1.ProgramNum.tomorrow,
            })}>
            {program.get('name')}
          </li>); }));
        }
    };
    StepTwoComponent.prototype.renderPackages = function () {
        var _a = this.props, packages = _a.packages, setPackageType = _a.setPackageType;
        if (!packages) {
            return null;
        }
        return (packages.map(function (pckg) {
            return (<li key={pckg.get('id')} className={classNames('options__item', {
                'is-active': pckg.get('isActive')
            })} onClick={function () { return setPackageType(pckg.get('id')); }}>
            {pckg.get('name')}
          </li>);
        }));
    };
    StepTwoComponent.prototype.render = function () {
        var _this = this;
        var currentPackage = this.props.currentPackage;
        return (<div>
        {this.renderHeader()}
        <div className='entry__box'>
          <ul className='options options--white wide mb30'>
            {this.renderPrograms()}
          </ul>
          <ul className='options options--white options--types wide mt30'>
            {this.renderPackages()}
          </ul>
        </div>
        <div className='entry__box'>
          <div className='entry-program-price'>{currentPackage ? currentPackage.get('cost') + ' руб' : ''}</div>
          <div className='input--btn'>
            <input ref={function (input) { return _this.promoText = input; }} type='text' name='promo' className='input__field' placeholder='Eсть промокод, вводи'/>
            <Button_1.Button onClick={function () { return _this.getPromo(); }} type='button' styleBtn={Button_1.ButtonTypes.info}>
              Применить
            </Button_1.Button>
          </div>
          <Button_1.Button onClick={function () { return _this.paymentCreate(); }} type='button' styleBtn={Button_1.ButtonTypes.success} wide={true}>
            Далее
          </Button_1.Button>
        </div>

        <div className='entry__link text-center'>
          <div className='entry-nav__item'>
            <react_router_dom_1.Link to='/' onClick={function () { return _this.logOut(); }}>Выйти</react_router_dom_1.Link>
          </div>
        </div>
      </div>);
    };
    return StepTwoComponent;
}(CustomSteps_1.default));
exports.StepTwoComponent = StepTwoComponent;
exports.StepTwo = react_redux_1.connect(mapStateToProps, {
    setChoosenProgram: ducks_1.setChoosenProgram,
    setPackageType: ducks_1.setPackageType,
    toggleModal: CustomModal_1.toggleModal,
    paymentCreate: ducks_1.paymentCreate,
    setPromo: ducks_1.setPromo,
    getPackage: ducks_1.getPackage,
    getPrograms: ducks_1.getPrograms
})(StepTwoComponent);
exports.default = ModalHOC_1.modalHOC(exports.StepTwo, ducks_1.moduleName, ducks_1.subModuleName);
