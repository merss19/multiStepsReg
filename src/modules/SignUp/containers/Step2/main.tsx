import * as React from 'react';
import {connect} from 'react-redux';
import {FormProps} from 'redux-form';
import CustomSteps, {CustomStepsProps} from '../CustomSteps';
import {Link} from 'react-router-dom';
import * as classNames from 'classnames';
import {storage} from 'tools/storage';
import {Steps, ProgramNum} from '../../const';
import {Program} from '../../interfaces';
import {Payment} from '../../interfaces';
import {Button, ButtonTypes} from 'components/Button';
import {toggleModal} from 'modules/CustomModal';
import {modalHOC} from 'components/HOC/ModalHOC';
import {
  setChoosenProgram,
  setPackageType,
  getPackage,
  getPrograms,
  setPromo,
  paymentCreate,
  moduleName,
  subModuleName
} from './ducks';
import {
  selectChoosenProgram,
  selectPromoError,
  selectPackageType,
  selectPackages,
  selectPrograms,
  selectPromo,
  selectPackage
} from '../../selectors';

interface OwnProps extends CustomStepsProps, FormProps<{}, {}, {}> {
  changeStep: (data: number) => void
  step: Steps
}

type ConnectedState = {
  choosenPackageType: number
  choosenProgram?: number
  packages: any
  currentPackage: any
  promo: string
  programs: Program[]
  promoError: string
}

type ConnectedDispatch = {
  setChoosenProgram(data: number): void
  setPackageType(data: number): void
  getPackage(data?: string): void
  getPrograms(): void
  paymentCreate(data: Payment, cb: () => void): void
  setPromo(data: string): void
  toggleModal(data: boolean): void
}

const mapStateToProps = (state: any): ConnectedState => ({
  programs: selectPrograms(state),
  choosenProgram: selectChoosenProgram(state),
  choosenPackageType: selectPackageType(state),
  packages: selectPackages(state),
  promo: selectPromo(state),
  currentPackage: selectPackage(state),
  promoError: selectPromoError(state)
});
type Props = OwnProps & ConnectedDispatch & ConnectedState

export class StepTwoComponent extends CustomSteps<Props> {

  promoText: any;

  componentWillMount() {
    const {programs, getPrograms, packages, setChoosenProgram, choosenProgram, getPackage} = this.props;
    if(!packages){
      getPackage();
    }
    if(!programs){
      getPrograms();
    }
  }
  componentWillReceiveProps(nextProps: Props){
    const {programs, choosenProgram, setChoosenProgram} = this.props
    if(nextProps.programs){
      if (!choosenProgram) {
        if(nextProps.programs){
          nextProps.programs.map((program: any) => {
            if (program.get('id') % ProgramNum.count === ProgramNum.hero) {
              setChoosenProgram(program.get('id'));
            }
          })
        }
      }
    }
  }
  shouldComponentUpdate(nextProps:Props){
    return (
      nextProps.step !== this.props.step
      ||  nextProps.packages !== this.props.packages
      ||  nextProps.programs !== this.props.programs
      ||  nextProps.choosenProgram !== this.props.choosenProgram
      ||  nextProps.choosenPackageType !== this.props.choosenPackageType
      ||  nextProps.promo !== this.props.promo
      ||  nextProps.currentPackage !== this.props.currentPackage
      ||  nextProps.promoError!== this.props.promoError
    );
  }

  getPromo() {
    const {getPackage, setPromo} = this.props;
    if (!this.promoText || !this.promoText.value.length) {
      return;
    }
    getPackage(this.promoText.value);
    setPromo(this.promoText.value);
  }

  paymentCreate() {
    const {choosenProgram, choosenPackageType, paymentCreate, promo} = this.props;
    let payload: Payment = {
      authToken: storage.load('token'),
      data: {
        program: choosenProgram,
        package: choosenPackageType,
        promoName:  promo
      }
    };
    paymentCreate(payload, () => this.resultAction());
  }

  resultAction() {
    const {changeStep, choosenProgram} = this.props;
    storage.save('program', choosenProgram,
      {path: '/', maxAge: 60 * 60 * 24 * 365 * 10});
    changeStep(Steps.three);
  }

  renderPrograms() {
    const {programs, setChoosenProgram} = this.props;
    if (!programs) {
      return <div>Нет программ</div>;
    } else {
      return (
        programs.map((program: any) => (
          <li key={program.get('id')}
              onClick={() =>{setChoosenProgram(program.get('id'))}}
              className={classNames('options__item', {
                'is-active': program.get('isActive'),
                'g-hero': program.get('id') % ProgramNum.count == ProgramNum.hero,
                'g-mather': program.get('id') % ProgramNum.count == ProgramNum.mother,
                'g-extreme': program.get('id') % ProgramNum.count == ProgramNum.extreme,
                'g-tomorrow': program.get('id') % ProgramNum.count == ProgramNum.tomorrow,
              })}
          >
            {program.get('name')}
          </li>)
        ))
    }
  }
  renderPackages() {
    const {packages, setPackageType} = this.props;
    if (!packages) {
      return null;
    }
    return (
      packages.map((pckg: any) => {
        return (
          <li
            key={pckg.get('id')}
            className={classNames('options__item', {
              'is-active': pckg.get('isActive')
            })}
            onClick={() => setPackageType(pckg.get('id'))}
          >
            {pckg.get('name')}
          </li>);
      })
    )
  }

  render() {
    const {currentPackage} = this.props;
    return (
      <div>
        {this.renderHeader()}
        <div className='entry__box'>
          <ul className='options options--white wide mb30'>
            {this.renderPrograms()}
          </ul>
          <ul className='options options--white options--types wide mt30'>
            {this.renderPackages()}
          </ul>
        </div>
        <div className='entry__box'>
          <div className='entry-program-price'>{currentPackage ? currentPackage.get('cost') + ' руб' : ''}</div>
          <div className='input--btn'>
            <input
              ref={input => this.promoText = input}
              type='text'
              name='promo'
              className='input__field'
              placeholder='Eсть промокод, вводи'
            />
            <Button
              onClick={() => this.getPromo()}
              type='button'
              styleBtn={ButtonTypes.info}
            >
              Применить
            </Button>
          </div>
          <Button
            onClick={() => this.paymentCreate()}
            type='button'
            styleBtn={ButtonTypes.success}
            wide={true}
          >
            Далее
          </Button>
        </div>

        <div className='entry__link text-center'>
          <div className='entry-nav__item'>
            <Link to='/' onClick={() => this.logOut()}>Выйти</Link>
          </div>
        </div>
      </div>);
  }
}

export const StepTwo = connect(
  mapStateToProps, {
    setChoosenProgram,
    setPackageType,
    toggleModal,
    paymentCreate,
    setPromo,
    getPackage,
    getPrograms
  })(StepTwoComponent);

export default modalHOC(StepTwo, moduleName, subModuleName);

