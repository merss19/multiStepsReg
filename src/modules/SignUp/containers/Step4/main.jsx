"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var react_router_dom_1 = require("react-router-dom");
var Button_1 = require("components/Button");
var CustomSteps_1 = require("../CustomSteps");
var StepFour = /** @class */ (function (_super) {
    __extends(StepFour, _super);
    function StepFour() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    StepFour.prototype.render = function () {
        var _this = this;
        return (<div className='entry__success'>
        <h3>Ваша заявка принята.</h3>
        <p>В ближайшее время мы с вами свяжемся!</p>
        <div className='text-center pt20'>
          <Button_1.Button type={'button'} styleBtn={Button_1.ButtonTypes.info}>
            <react_router_dom_1.Link to='/' onClick={function () { return _this.logOut(); }}>Вернуться на главный сайт</react_router_dom_1.Link>
          </Button_1.Button>
        </div>
      </div>);
    };
    return StepFour;
}(CustomSteps_1.default));
exports.StepFour = StepFour;
exports.default = StepFour;
