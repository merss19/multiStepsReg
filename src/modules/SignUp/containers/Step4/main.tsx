import * as React from 'react';
import {Link} from 'react-router-dom';
import {Button, ButtonTypes} from 'components/Button';
import CustomSteps, {CustomStepsProps} from '../CustomSteps';

interface OwnProps extends CustomStepsProps {}
export class StepFour extends CustomSteps<OwnProps> {
  render() {
    return (
      <div className='entry__success'>
        <h3>Ваша заявка принята.</h3>
        <p>В ближайшее время мы с вами свяжемся!</p>
        <div className='text-center pt20'>
          <Button
            type={'button'}
            styleBtn={ButtonTypes.info}
          >
            <Link to='/' onClick={() => this.logOut()}>Вернуться на главный сайт</Link>
          </Button>
        </div>
      </div>
    );
  }
}

export default StepFour;
