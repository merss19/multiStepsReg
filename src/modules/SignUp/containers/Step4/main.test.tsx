import {shallow, mount} from 'enzyme';
import {StepFour} from './main';
import * as React from 'react';
import toJson from 'enzyme-to-json';
import * as sinon from 'sinon';
import {Provider} from 'react-redux';
import configureStore from '../../../../store';
import {BrowserRouter as Router, Link} from 'react-router-dom';
import {configure} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import {storage} from '../../../../tools/storage/index';
import {Steps} from '../../const';
import {Button, ButtonTypes} from "../../../../components/Button/button";
import * as Immutable from 'immutable';

const store = configureStore();
configure({adapter: new Adapter()});

let
  match,
  location,
  changeStep,
  history;
const start = () => {
  beforeEach(() => {
    changeStep = jest.fn(),
    match = {
      isExact: true,
      params: {},
      path: "/task",
      url: "/task"
    },
    location = {},
    history = {
      replace: jest.fn()
    }
  });
  const props = {
    match: match,
    changeStep: changeStep,
    location: location,
    history: history
  };
  const buildComponent = () => {
    return shallow(<StepFour {...props}/>)
  };
  const buildComponentMount = () => {
    return mount(
      <Provider store={store}>
        <Router>
          <StepFour {...props}/>
        </Router>
      </Provider>)
  };
  return {
    buildComponentMount,
    props,
    buildComponent
  }
};
describe('Component StepThree', () => {
  let component;
  start();
  beforeEach(() => {
    component = start().buildComponent();
  });
  it('renders component', () => {
    expect(component).toHaveLength(1);
  });
  it('renders component Mount', () => {
    const component = start().buildComponentMount();
    expect(component).toHaveLength(1);
  });
  it('renders correctly', () => {
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
  it('click Link', () => {
    const spy = sinon.spy(StepFour.prototype, 'logOut');
    const button = component.find('Link');
    button.simulate('click');
    expect(spy.called).toBe(true);
  });
});


