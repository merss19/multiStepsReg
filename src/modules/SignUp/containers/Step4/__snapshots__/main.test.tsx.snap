// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component StepThree renders correctly 1`] = `
<div
  className="entry__success"
>
  <h3>
    Ваша заявка принята.
  </h3>
  <p>
    В ближайшее время мы с вами свяжемся!
  </p>
  <div
    className="text-center pt20"
  >
    <Component
      prefix="btn"
      styleBtn="secondary"
      type="button"
      wide={false}
    >
      <Link
        onClick={[Function]}
        replace={false}
        to="/"
      >
        Вернуться на главный сайт
      </Link>
    </Component>
  </div>
</div>
`;
